src/libflake/flake/flake.cc

```diff
@@ -783,7 +783,7 @@ LockedFlake lockFlake(
                             auto relPath = (topRef.subdir == "" ? "" : topRef.subdir + "/") + "flake.lock";
                             auto outputLockFilePath = *sourcePath / relPath;
 
-                            bool lockFileExists = pathExists(outputLockFilePath);
+                            bool lockFileExists = std::filesystem::exists(outputLockFilePath);
```

Comment:
`fs::exists_symlnk` will avoid dereferencing symlinks, FYI.

Reply:
I don't think that exists. There is `std::filesystem::exists(path.symlink_status())` though.

Reply:
It does exist, I made it.

Reply:
https://github.com/NixOS/nix/blob/4adb579c428963886e2bcc7397f242a500cdf6d8/src/libutil/file-system.hh#L134-L149